Patch will fix compile problems with gcc-4.1.1
Joerg Bornkessel hd_brummy@gentoo.org 
05 Sep 2006

written with help of LordJaxom

diff -Naur setup-0.3.1.orig/menus.h setup-0.3.1/menus.h
--- setup-0.3.1.orig/menus.h	2006-09-05 20:35:37.941397850 +0200
+++ setup-0.3.1/menus.h	2006-09-05 20:39:08.866579848 +0200
@@ -101,7 +101,7 @@
   public:
     MenuEntryValueList();
     ~MenuEntryValueList();
-    MenuEntryValueList const &MenuEntryValueList::operator =(MenuEntryValueList const &right   );
+    MenuEntryValueList const &operator =(MenuEntryValueList const &right   );
     void copy(MenuEntryValueList const &other);
     void destroy();
     void Add(const char *value);
@@ -132,9 +132,9 @@
     MenuEntry();
     ~MenuEntry();
     void destroy();
-    bool MenuEntry::AddEntry( TiXmlNode *node  );
-    bool MenuEntry::AddSubEntry( TiXmlNode *node  );
-    bool MenuEntry::AddSelectionValues( TiXmlNode *node, const char *selection);
+    bool AddEntry( TiXmlNode *node  );
+    bool AddSubEntry( TiXmlNode *node  );
+    bool AddSelectionValues( TiXmlNode *node, const char *selection);
     TiXmlNode *SaveXml( );
     void SetSysconfigName(const char* nam);
     const char*GetSysconfigName();
diff -Naur setup-0.3.1.orig/plugins.h setup-0.3.1/plugins.h
--- setup-0.3.1.orig/plugins.h	2006-09-05 20:35:37.953398599 +0200
+++ setup-0.3.1/plugins.h	2006-09-05 20:36:53.458117352 +0200
@@ -33,20 +33,20 @@
     bool operator< (Plugin &param);
     bool operator> (Plugin &param);
     void Print();
-    const char* Plugin::GetName();
-    const char* Plugin::GetInfo();
-    const char* Plugin::GetParameter();
-    void Plugin::SetName(const char* name);
-    void Plugin::SetInfo(const char* info);
-    void Plugin::SetParameter(const char* parameter);
-    void Plugin::SetActive(int flag);
-    void Plugin::SetProtect(bool protect);
-    int  Plugin::GetActive();
-    int  *Plugin::GetActiveRef();
-    void Plugin::SetInSystem(bool flag);
-    bool Plugin::GetInSystem();
-    bool Plugin::GetProtect();
-    const char* Plugin::GetActiveString();
+    const char* GetName();
+    const char* GetInfo();
+    const char* GetParameter();
+    void SetName(const char* name);
+    void SetInfo(const char* info);
+    void SetParameter(const char* parameter);
+    void SetActive(int flag);
+    void SetProtect(bool protect);
+    int  GetActive();
+    int  *GetActiveRef();
+    void SetInSystem(bool flag);
+    bool GetInSystem();
+    bool GetProtect();
+    const char* GetActiveString();
 };
 
 class Plugins  : public cList<Plugin> 
